homeassistant/components/derivative/config_flow.py

```diff
@@ -44,6 +44,11 @@
 
 OPTIONS_SCHEMA = vol.Schema(
     {
+        vol.Required(CONF_SOURCE): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[COUNTER_DOMAIN, INPUT_NUMBER_DOMAIN, SENSOR_DOMAIN]
+            ),
```

Comment:
I think this need to maintain same unit to allow it to be changed. See here: https://github.com/home-assistant/core/pull/116740/files#diff-d6bb8150a2868b6dd04145e6bfcc804b99e27e9f2abf100c8483ff60bcaf5bbfR60

Reply:
Done!